### PR TITLE
PLT-4492 Mentions with a comma after them don't trigger notification jewel

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
The `getExplicitMentions` was spliting the message first by a space and tried to figure out which word was a mention for users in the channel and outside the channel, if none where found then it reitereted again but spliting each word using a list of characters to try and find the mentions.

what I did is I removed the first check for each word and now we are splitting the message by each rune on the `SplitRunes` map removing the need to iterate twice

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4492

### CODE REVIEW

**Comment by crspeller** [comment]
@hmhealey Please take a look.

File: api/post.go [hunk 1]
```diff
@@ -486,53 +486,27 @@ func getExplicitMentions(message string, keywords map[string][]string) (map[stri
...
 
-	for _, word := range strings.Fields(message) {
-		isMention := false
+	splitWords := strings.FieldsFunc(message, func(c rune) bool {
+		return model.SplitRunes[c]
+	})
 
-		if word == "@here" {
+	for _, splitWord := range splitWords {
+		l4g.Debug("split word = %v", splitWord)
```
Comment (hmhealey) [inline api/post.go]: You forgot to remove a debug line :)

File: api/post.go [hunk 2]
```diff
@@ -486,53 +486,27 @@ func getExplicitMentions(message string, keywords map[string][]string) (map[stri
...
 	}
 
-	for _, word := range strings.Fields(message) {
-		isMention := false
+	splitWords := strings.FieldsFunc(message, func(c rune) bool {
+		return model.SplitRunes[c]
+	})
 
-		if word == "@here" {
+	for _, splitWord := range splitWords {
```
Comment (hmhealey) [inline api/post.go]: Does this still let you mention users who's names contain a period, underscore, or hyphen? That was the reason we had that weird, two-step splitting process

File: api/post.go (hunk 1 above)
Comment (enahum) [inline api/post.go]: Hmm nonit doesnt, i'll come up with another fix for this

**Pull request merged**